### PR TITLE
[ML] Exceptions about starting native processes now include the node

### PR DESCRIPTION
**Pull request by droberts195**
Previously exceptions about starting native processes did not
include the name of the node where the attempt to start the
process failed.

Although this doesn't matter when looking at the log of the
node where the attempt was made, it is crippling when the
report of the problem is just the exception received by a
client. When the initial report comes from a client exception
we need to be able to easily determine which node to ask for
the logs for, or to look at operating system level issues on.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java [hunk 1]
```diff
@@ -57,6 +58,7 @@ public NativeAnalyticsProcessFactory(Environment env,
                                          DataFrameAnalyticsAuditor auditor) {
         this.env = Objects.requireNonNull(env);
         this.nativeController = Objects.requireNonNull(nativeController);
+        this.clusterService = Objects.requireNonNull(clusterService);
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java:61]: Seems to me that we don't need to keep a reference to the clusterService. we could have `this.nodeName = Objects.requireNonNull(clusterService).getNodeName()` as the internal `nodeName` is a `final` in the `ClusterService` object.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java [hunk 2]
```diff
@@ -125,11 +127,11 @@ private void createNativeProcess(String jobId, AnalyticsProcessConfig analyticsP
...
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
-            LOGGER.warn("[{}] Interrupted while launching data frame analytics process", jobId);
+            logger.warn("[{}] Interrupted while launching data frame analytics process", jobId);
         } catch (IOException e) {
             String msg = "[" + jobId + "] Failed to launch data frame analytics process";
-            LOGGER.error(msg);
-            throw ExceptionsHelper.serverError(msg, e);
+            logger.error(msg);
+            throw ExceptionsHelper.serverError(msg + " on [" + clusterService.getNodeName() + "]", e);
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java:134]:
```suggestion
            throw ExceptionsHelper.serverError("{} on [{}]", e, msg, clusterService.getNodeName());
```
or simply `nodeName` if previous suggestions are accepted.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java [hunk 3]
```diff
@@ -37,12 +37,13 @@
...
 public class NativeAnalyticsProcessFactory implements AnalyticsProcessFactory<AnalyticsResult> {
 
-    private static final Logger LOGGER = LogManager.getLogger(NativeAnalyticsProcessFactory.class);
+    private static final Logger logger = LogManager.getLogger(NativeAnalyticsProcessFactory.class);
 
     private static final NamedPipeHelper NAMED_PIPE_HELPER = new NamedPipeHelper();
 
     private final Environment env;
     private final NativeController nativeController;
+    private final ClusterService clusterService;
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java:46]:
```suggestion
    private final String nodeName;
```

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java (hunk 1 above)
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java:61]:
```suggestion
        this.nodeName = Objects.requireNonNull(clusterService).getNodeName();
```

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeMemoryUsageEstimationProcessFactory.java [hunk 4]
```diff
@@ -33,18 +33,20 @@
...
 public class NativeMemoryUsageEstimationProcessFactory implements AnalyticsProcessFactory<MemoryUsageEstimationResult> {
 
-    private static final Logger LOGGER = LogManager.getLogger(NativeMemoryUsageEstimationProcessFactory.class);
+    private static final Logger logger = LogManager.getLogger(NativeMemoryUsageEstimationProcessFactory.class);
 
     private static final NamedPipeHelper NAMED_PIPE_HELPER = new NamedPipeHelper();
 
     private final Environment env;
     private final NativeController nativeController;
+    private final ClusterService clusterService;
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeMemoryUsageEstimationProcessFactory.java:42]: All suggestions in `NativeAnalyticsProcessFactory.java` apply here as well.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/NativePyTorchProcessFactory.java [hunk 5]
```diff
@@ -37,13 +37,15 @@
 
     private final Environment env;
     private final NativeController nativeController;
+    private final ClusterService clusterService;
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/NativePyTorchProcessFactory.java:40]: ditto, I think we only need the `nodeName`

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/normalizer/NativeNormalizerProcessFactory.java [hunk 6]
```diff
@@ -28,17 +28,19 @@
 
 public class NativeNormalizerProcessFactory implements NormalizerProcessFactory {
 
-    private static final Logger LOGGER = LogManager.getLogger(NativeNormalizerProcessFactory.class);
+    private static final Logger logger = LogManager.getLogger(NativeNormalizerProcessFactory.class);
     private static final NamedPipeHelper NAMED_PIPE_HELPER = new NamedPipeHelper();
 
     private final Environment env;
     private final NativeController nativeController;
+    private final ClusterService clusterService;
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/normalizer/NativeNormalizerProcessFactory.java:36]: we should only need the node name.

**Review by benwtrent — COMMENTED** [review]
Include node name is good here. Just some comments on logging formatting + clusterService references.

**Review by benwtrent — APPROVED** [review]

**Comment by droberts195** [comment]
Jenkins run elasticsearch-ci/part-2

**Comment by droberts195** [comment]
@elasticmachine update branch

**Pull request merged**